modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PinMeshGenerator.h"
+
+#include <cmath>
+#include "MooseApp.h"
+#include "Factory.h"
+#include "libmesh/elem.h"
+
+registerMooseObject("ReactorApp", PinMeshGenerator);
+
+InputParameters
+PinMeshGenerator::validParams()
+{
+  auto params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>(
+      "reactor_params",
+      "The ReactorMeshParams MeshGenerator that is the basis for this component conformal mesh.");
+
+  params.addRequiredParam<subdomain_id_type>("pin_type",
+                                             "The integer ID for this pin type definition");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "pitch", "pitch>0.0", "The pitch for the outermost boundary polygon");
+
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "num_sectors", "num_sectors>0", "Number of azimuthal sectors in each quadrant");
+
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "ring_radii", "ring_radii>0.0", "Radii of major concentric circles");
```

Comment:
```suggestion
      "ring_radii", "ring_radii>0.0", "Radii of major concentric circles of the pin");
```
just to say that it s not for every region.